### DISCUSSION TITLE
Redown load peloton json files not working

### DISCUSSION THREAD

**Discussion by rafischer54** [philosowaffle/peloton-to-garmin#203] [Q&A]
I have not been saving the peloton json files but there is a lot of info there I'd like to convert to csv files and analyze.  Since I have not been saving the peloton json files I renamed the sync  file, increased by files to download to 2000, disabled the upload to garmin and ran the peloton to garmin console.  To my surprise over a 100 workout were not downloaded.  Ran the program again and no new json files were created.  I'd like to recover these missing json files, any ideas why all peloton workout json files are not being crreated?  There does not seem to be any patterns to the missing json files.  Thx

**Comment by philosowaffle** [comment]
Not any ideas without knowing more about which workouts were missing and what was different about them.

You can try setting the [log level](https://philosowaffle.github.io/peloton-to-garmin/configuration/json.html#serilog-config) to `Debug` and running the job again.

 Then search through the log output for messages like:
1. Messages beginning with `Skipping`
2. `Total workouts found after filtering InProgress and ExcludeWorkoutTypes:`
3. Messages beginning with `Failed to deserialize`

**Reply by rafischer54** [reply]
Thx for the quick response but user error!!  I demoed the program to my son so I had a bunch of extra workouts that did not belong to me.  Re-downloading the files cleaned all of that up.  The raw Peloton json files has a wealth of information that I've converted to csv files so I can compare rides that I have taken more than once to see the improvement.

Again, thx for the quick replay and sorry for the false alarm!!
